oneview_redfish_toolkit/blueprints/zone.py

```diff
@@ -40,17 +40,45 @@ def get_zone(uuid):
     """
     profile_template = g.oneview_client.server_profile_templates.get(uuid)
 
-    encl_group_uri = profile_template['enclosureGroupUri']
-    sh_type_uri = profile_template['serverHardwareTypeUri']
+    logical_enclosure_resource = \
+        _get_logical_enclosure_by_spt(profile_template)
+    enclosures_uris = logical_enclosure_resource["enclosureUris"]
+    server_hardware_filter = _create_server_hardware_filter(enclosures_uris)
 
-    available_targets = g.oneview_client\
-        .server_profiles\
-        .get_available_targets(enclosureGroupUri=encl_group_uri,
-                               serverHardwareTypeUri=sh_type_uri)
+    server_hardware_list = g.oneview_client.server_hardware.get_all(
+        filter=server_hardware_filter
+    )
 
     drives = g.oneview_client.index_resources \
         .get_all(category="drives", count=10000)
 
-    zone_data = Zone(profile_template, available_targets, drives)
+    zone_data = Zone(profile_template, server_hardware_list, drives)
 
     return ResponseBuilder.success(zone_data)
+
+
+def _create_server_hardware_filter(enclosures_uris):
+    enclosure_filters = list()
+    location_uri = "'locationUri'='{}'"
+
+    for enclosure_uri in enclosures_uris:
+        enclosure_filters.append(location_uri.format(enclosure_uri))
+
+    server_hardware_filter = ' OR '.join(enclosure_filters)
+
+    return server_hardware_filter
+
+
+def _get_logical_enclosure_by_spt(profile_template):
+    encl_group_uri = profile_template['enclosureGroupUri']
+    encl_group_index_trees_uri = "/rest/index/trees{}?childDepth=1"
```

Comment:
Could we use a childDepth=2, and get directly the enclosure URIs under the "LOGICAL_ENCLOSURE_TO_ENCLOSURE" of the children on the second depth?